### PR TITLE
pm/fm: defensively harden parsing edge cases

### PR DESCRIPTION
**Pull request by rgerhards**
## Summary
- defensively harden pmdb2diag, pmsnare, pmciscoios, pmpanngfw, pmrfc3164, and fmhttp edge cases found during the pm/fm audit pass
- add regression tests for malformed parser tails and force.tagEndingByColon no-colon handling

## Validation
- PASS: Ubuntu 26.04 static analyzer container, scan-build reported no bugs
- PASS: targeted local tests: pmciscoios_truncated_input.sh, pmsnare_malformed_tag_tail.sh, pmdb2diag_malformed_tail.sh, pmrfc3164_force_tag_no_colon.sh
- PASS: representative existing tests: pmsnare-default.sh, pmrfc3164-tagEndingByColon.sh, pmrfc3164-defaultTag.sh, rscript_http_request.sh, pmdb2diag_parse.sh
- PASS: valgrind for the four new regression tests
- PASS: make distcheck TEST_RUN_TYPE=MOCK-OK
- BLOCKED: full Ubuntu 26.04 container CI still has 1 unrelated failure in imuxsock_unlink_off_stale_socket.sh; the expected stale-socket diagnostic was emitted on stderr, but the test expected it in the .started file, which was not created

## Notes
mmpstrucdata was intentionally not touched.

### CODE REVIEW

File: contrib/pmdb2diag/pmdb2diag.c [hunk 1]
```diff
@@ -78,21 +83,26 @@ ENDisCompatibleWithFeature
...
     assert(pMsg->pszRawMsg != NULL);
 
     DBGPRINTF("Message will now be parsed by \"db2diag\" parser.\n");
-    if (pMsg->iLenRawMsg - (int)pMsg->offAfterPRI < pInst->levelpos + 4) ABORT_FINALIZE(RS_RET_COULD_NOT_PARSE);
+    rawStart = (char *)pMsg->pszRawMsg + pMsg->offAfterPRI;
+    rawLen = pMsg->iLenRawMsg - (int)pMsg->offAfterPRI;
+    if (rawLen <= 0 || pInst->levelpos < 0 || pInst->timepos < 0 || pInst->pidstarttoprogstartshift < 0) {
+        ABORT_FINALIZE(RS_RET_COULD_NOT_PARSE);
+    }
+    if (rawLen < pInst->levelpos + 4 || rawLen <= pInst->timepos) ABORT_FINALIZE(RS_RET_COULD_NOT_PARSE);
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply contrib/pmdb2diag/pmdb2diag.c]: Done. I changed the pmdb2diag checks to compare levelpos against the remaining raw length before adding constants or level-specific lengths, and added a guard before applying pidstarttoprogstartshift.

File: plugins/fmhttp/fmhttp.c [hunk 2]
```diff
@@ -104,14 +112,12 @@ static void ATTR_NONNULL() doFunc_http_request(struct cnffunc *__restrict__ cons
     }
 
 
-    CHKmalloc(ret->d.estr = es_newStrFromCStr(curlData->reply, curlData->replyLen));
+    CHKmalloc(ret->d.estr = es_newStrFromCStr(curlData.reply, curlData.replyLen));
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply plugins/fmhttp/fmhttp.c:117]: Done. I capped the accumulated HTTP reply before conversion to libestr by rejecting replies that would exceed UINT_MAX, matching es_size_t on supported libestr headers.

[the remark this answers is not in the document]
Comment (rgerhards) [reply plugins/fmhttp/fmhttp.c:117]: Done. The same fmhttp update now rejects accumulated replies that would exceed UINT_MAX before calling es_newStrFromCStr, avoiding the narrowing conversion into es_size_t.
